altapay.php

```diff
@@ -2464,21 +2350,41 @@ private function getVoucherDetails()
     /**
      * Returns array of cart rule discounts applied on each product from created order
      *
-     * @param $order
+     * @param array $order
      *
-     * @return int|mixed
+     * @return array
      */
     private function getCartRuleDiscounts($order)
     {
-        $cartRuleDiscounts = 0;
+        $cartRuleDiscounts = [];
         $discountPercent   = reset(Db::getInstance()->executeS('SELECT * FROM ' . _DB_PREFIX_ . 'altapay_cartInfo WHERE id_cart = ' . $order->id_cart));
 
-
         if (!empty($discountPercent)) {
             $discountPercent   = json_decode($discountPercent['productDetails']);
```

Comment:
We need to handle cases where json_decode fails. Previously the outer functions would handle this was handled by the caller by doing nothing if the result was falsy. We can maintain this behavior by returning an empty array, or we could be more explicit and throw an exception.